### PR TITLE
powershell: fix prepend for new variables and fix unsetenv for non-existing variables

### LINKED ISSUE #1476: addressing missing test for powershell prependenv / unset
Token issue to create issue against, to bundle up things relating to:
https://github.com/AcademySoftwareFoundation/rez/issues/1255
and
https://github.com/AcademySoftwareFoundation/rez/pull/1434

### PR DESCRIPTION
**Pull request by maxnbk**
Closes #1476
Supercedes PR by @SitiSchu on #1434 (but includes his commit cherrypicked to maintain history/credit)

The test, and commits, also address _the same issue_ but in the "unset" logic, because that too is a victim of this same issue, which is validated by the test, and was in fact the reason I came complaining to @instinct-vfx about not understanding something I was seeing, I just wasn't bright enough to understand it right away.

### CODE REVIEW

**Comment by JeanChristopheMorinPerso** [comment]
@maxnbk FYI, since you are admin on the repo, you could have pushed your commit into @SitiSchu 's PR.

**Comment by maxnbk** [comment]
> @maxnbk FYI, since you are admin on the repo, you could have pushed your commit into @SitiSchu 's PR.

How? The branch exists on their fork.

**Comment by JeanChristopheMorinPerso** [comment]
If the contributor checks the "Allow maintainers to edit PR", the contributor grants the admins/maintainers access to push to their branch in their fork. See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests.

> On user-owned forks, if you want to allow anyone with push access to the upstream repository to make changes to your pull request, select Allow edits from maintainers.

**Comment by maxnbk** [comment]
> If the contributor checks the "Allow maintainers to edit PR", the contributor grants the admins/maintainers access to push to their branch in their fork. See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests.
> 
> > On user-owned forks, if you want to allow anyone with push access to the upstream repository to make changes to your pull request, select Allow edits from maintainers.

Gotcha, good to know.

**Review by JeanChristopheMorinPerso — APPROVED** [review]
LGTM! Thanks @maxnbk and @SitiSchu !

File: src/rezplugins/shell/_utils/powershell_base.py [hunk 1]
```diff
@@ -280,7 +280,9 @@ def appendenv(self, key, value):
             .format(key, os.path.pathsep, value))
 
     def unsetenv(self, key):
-        self._addline(r"Remove-Item Env:\%s" % key)
+        self._addline(
+            'Remove-Item -ErrorAction -SilentlyContinue "Env:{0}"'.format(key)
```
Comment (herronelou) [inline src/rezplugins/shell/_utils/powershell_base.py:284]: I think there's an extra dash here that shouldn't be there, I got this:
```
Line |
 913 |  Remove-Item -ErrorAction -SilentlyContinue "Env:CUDA_CACHE_MAXSIZE"
     |              ~~~~~~~~~~~~
     | Cannot bind parameter 'ErrorAction'. Cannot convert value "-SilentlyContinue" to type
     | "System.Management.Automation.ActionPreference". Error: "Unable to match the identifier name -SilentlyContinue
     | to a valid enumerator name. Specify one of the following enumerator names and try again: SilentlyContinue, Stop,
     | Continue, Inquire, Ignore, Suspend, Break"

Comment (SitiSchu) [reply src/rezplugins/shell/_utils/powershell_base.py:284]: Yeah it should just be `SilentlyContinue` without the hyphen

**Comment by JeanChristopheMorinPerso** [comment]
Thanks @maxnbk and @SitiSchu !

**Review by instinct-vfx — COMMENTED** [review]
LGTM looks good. Thanks @SitiSchu  and @maxnbk

**Pull request merged**